src/main/java/org/folio/circulation/resources/SlipsResource.java

```diff
@@ -100,50 +105,77 @@ private void getMany(RoutingContext routingContext) {
       .thenComposeAsync(r -> r.after(ctx -> new RequestFetchService().fetchRequests(ctx, clients, requestType)))
       .thenComposeAsync(r -> r.after(ctx -> userRepository.findUsersForRequests(
         ctx.getRequests())))
-      .thenComposeAsync(result -> result.after(patronGroupRepository::findPatronGroupsForRequestsUsers))
+      .thenComposeAsync(r -> r.after(patronGroupRepository::findPatronGroupsForRequestsUsers))
       .thenComposeAsync(r -> r.after(departmentRepository::findDepartmentsForRequestUsers))
       .thenComposeAsync(r -> r.after(addressTypeRepository::findAddressTypesForRequests))
       .thenComposeAsync(r -> r.after(servicePointRepository::findServicePointsForRequests))
       .thenApply(flatMapResult(this::mapResultToJson))
       .thenComposeAsync(r -> r.combineAfter(() -> servicePointRepository.getServicePointById(servicePointId),
-        this::addPrimaryServicePointNameToStaffSlipContext))
-      .thenApply(r -> r.map(JsonHttpResponse::ok))
-      .thenAccept(context::writeResultToHttpResponse);
+        this::addPrimaryServicePointNameToStaffSlipContext));
   }
 
-  private Result<Object> returnNoRecordsIfSearchSlipsDisabled(PrintHoldRequestsConfiguration config,
-    WebContext context) {
-
-    if (config == null || !config.isPrintHoldRequestsEnabled()) {
-      log.info("returnNoRecordsIfSearchSlipsDisabled:: Print hold requests configuration is disabled");
-      context.writeResultToHttpResponse(succeeded(JsonHttpResponse.ok(
-        new io.vertx.core.json.JsonObject()
-          .put(SEARCH_SLIPS_KEY, new JsonArray())
-          .put(TOTAL_RECORDS_KEY, 0)
-      )));
+  private CompletableFuture<Result<Boolean>> isStaffSlipsPrintingDisabled(Clients clients) {
+    final var configurationRepository = new ConfigurationRepository(clients);
+    final var circulationSettingsRepository = new CirculationSettingsRepository(clients);
+
+    if (PICK_SLIPS_KEY.equals(collectionName) && requestType == RequestType.PAGE) {
+      log.info("returnEmptyRecordsIfNeeded: PICK_SLIPS_KEY and PAGE requestType condition met");
+      return circulationSettingsRepository.findBy(PRINT_EVENT_FLAG_QUERY)
+        .thenApply(r -> r.next(this::isPrintingPickSlipsDisabled));
+    } else if (SEARCH_SLIPS_KEY.equals(collectionName) && requestType == RequestType.HOLD) {
+      log.info("returnEmptyRecordsIfNeeded: SEARCH_SLIPS_KEY and HOLD requestType condition met");
+      return configurationRepository.lookupPrintHoldRequestsEnabled()
+        .thenApply(r -> r.next(config -> succeeded(!config.isPrintHoldRequestsEnabled())));
+    } else {
+      return ofAsync(false);
     }
-    return succeeded(null);
+  }
+
+  private Result<Boolean> isPrintingPickSlipsDisabled(
+    MultipleRecords<CirculationSetting> settingsRecords) {
+
+    log.debug("isPrintingPickSlipsEnabled:: parameters settingsRecords: {}",
+      () -> multipleRecordsAsString(settingsRecords));
+
+    return succeeded(!Optional.ofNullable(settingsRecords)
```

Comment:
No need to wrap return value into a `Result`. Also, I think returning immediately if `settingsRecords` is null will make code slightly more readable:

```
if (settingsRecords == null) {
      return false;
    }

    return settingsRecords.getRecords()
      .stream()
      .filter(setting -> PRINT_EVENT_LOG_FEATURE.equals(setting.getName()))
      .findFirst()
      .map(CirculationSetting::getValue)
      .map(setting -> !setting.getBoolean("enablePrintLog"))
      .orElse(false);
 ```